<?xml version="1.0"?>
<launch>
    <!-- Default blank to do live calibration -->
    <arg name="file_path" default=""/>

    <!-- Init pose of the arm -->
    <arg name="init_x"  default="0.0"/>
    <arg name="init_y"  default="0.65"/>
    <arg name="init_z"  default="0.35"/>

    <arg name="planning_link" default="base_link"/>

    <arg name="platform_server" default="server"/>

    <arg name="sensor" default=""/>
    <!-- Stereo camera topics -->
    <!-- NOTE: precalibration we only have left and right images to work with -->
    <arg name="sensor_topics" default="$(find cares_hand_eye_calibration)/config/stereo_calibration.yaml"/>

    <arg name="display" default="true"/>

    <!-- The input reference frames -->
    <arg name="robot_base_frame"     default="base_link" />
    <arg name="robot_effector_frame" default="stereo_pair/base_link" />
    <arg name="tracking_base_frame"  default="stereo_pair/left_frame" />

    <!-- Which control frame will be used to move the sensor around during calibration -->
    <arg name="robot_control_frame"  default="$(arg robot_effector_frame)" />

    <!--Charuco Board Parameters-->
    <arg name="board_width"   default="8"/><!--Number of corners-->
    <arg name="board_height"  default="5"/><!--Number of corners-->

    <arg name="square_length" default="45"/><!--mm-->
    <arg name="marker_length" default="23.0"/><!--mm-->

    <arg name="image_width"  default="1936"/>
    <arg name="image_height" default="1216"/>
    
    <!--1 Charuco-->
    <arg name="dictionary" default="3"/>
    <arg name="method"     default="1"/>

    <!-- Aruco Diamond IDs -->
    <arg name="centre"    default="11"/>
    <arg name="top_left"  default="6"/>
    <arg name="top_right" default="7"/>
    <arg name="bot_right" default="16"/>
    <arg name="bot_left"  default="15"/>

    <include file="$(find cares_hand_eye_calibration)/launch/auto_calibration.launch" pass_all_args="true"> 
    </include>
    
    <!-- Stereo Diamond Detector Service -->     
    <include file="$(find aruco_detector)/launch/stereo_diamond_detector_service.launch" pass_all_args="true">
    </include>

    <!-- Stereo Calibration Service -->
    <include file="$(find stereo_calibration)/launch/service_stereo_calibration.launch" pass_all_args="true">
    </include>    
</launch>
